pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -58,16 +58,25 @@ func (a *Operator) syncOperatorGroups(obj interface{}) error {
 	a.Log.Debug("Cluster roles completed")
 
 	for _, csv := range a.csvSet(op.Namespace, v1alpha1.CSVPhaseAny) {
-		origCSVannotations := csv.GetAnnotations()
+		origCSVannotations := a.copyOperatorGroupAnnotations(&csv.ObjectMeta)
```

Comment:
why copy out the just the three annotations and then immediately set just those three below with `addOperatorGroupAnnotations`? If we aren't going to retain the other annotation values, then we can just directly build the set of annotations here and not use these `copyOperatorGroupAnnotations`/`addOperatorGroupAnnotations` methods?

Reply:
The idea here wasn't to change the logic, but to fix the bug of not detecting annotation changes due to getting a reference instead of a copy. The new `copyOperatorGroupAnnotations` method ensures a copy is done so that the reflection test below works properly.